src/metadata.rs

```diff
@@ -4,12 +4,16 @@ use std::ops::{Deref, DerefMut};
 use super::Error;
 use crate::Error::MalformedMetadata;
 
-#[derive(Debug)]
-pub struct MetadataMap<'a>(HashMap<&'a str, &'a str>);
+#[derive(Debug, Clone)]
 
-impl<'a> MetadataMap<'a> {
+/// Define a struct called `MetadataMap` that represents a map of metadata.
+/// It is a simple wrapper around a `HashMap` with `String` keys and `String` values.
+/// This struct is used to store key-value pairs of metadata associated with a translation entry or other data.
+pub struct MetadataMap(HashMap<String, String>);
+
+impl MetadataMap {
     /// Returns a string that indicates the character set.
-    pub fn charset(&self) -> Option<&'a str> {
+    pub fn charset(&self) -> Option<&str> {
```

Comment:
Are these lifetime changes still a requirement?

Reply:
@rmccue Yes, they are still a requirement because we are adding the MetadataMap type to the Catalog struct and the Catalog is meant to return an owned value. If we keep the lifetimes on the MetadataMap implementation then we will need to add a lifetime to the Catalog struct hence making the data in the Catalog local to the function which will throw an error when we try to return that in `Ok(catalog)` https://github.com/rmccue/gettext-rs/blob/02a52358c86d9dc10ddb0273816921e4320f4522/src/parser.rs#L167 because it would be referencing temporary data that is already deallocated when `Ok(catalog)` is run.